cpp/velox/compute/WholeStageResultIterator.cc

```diff
@@ -551,8 +551,13 @@ std::unordered_map<std::string, std::string> WholeStageResultIterator::getQueryC
 
     configs[velox::core::QueryConfig::kSparkPartitionId] = std::to_string(taskInfo_.partitionId);
 
-    // Enable Spark legacy date formatter if spark.sql.legacy.timeParserPolicy is set to 'LEGACY'.
-    if (veloxCfg_->get<std::string>(kSparkLegacyTimeParserPolicy, "") == "LEGACY") {
+    // Enable Spark legacy date formatter if spark.sql.legacy.timeParserPolicy is set to 'LEGACY'
+    // or 'legacy'.
+    auto sparkPolicy = veloxCfg_->get<std::string>(kSparkLegacyTimeParserPolicy, "");
+    for (auto& c : sparkPolicy) {
+      c = std::toupper(static_cast<unsigned char>(c));
+    }
+    if (sparkPolicy == "LEGACY") {
```

Comment:
It seems this is a common issue while passing config from jvm to c++, maybe we should done this in GlutenConfig while `getNativeSessionConf`